src/SecurityManager.js

```diff
@@ -66,6 +77,20 @@ async function confirmToDismiss() {
     return !sure;
 }
 
+function makeInputToKey(keyInfo) {
+    return async ({ passphrase, recoveryKey }) => {
+        if (passphrase) {
+            return deriveKey(
+                passphrase,
+                keyInfo.passphrase.salt,
+                keyInfo.passphrase.iterations,
+            );
+        } else {
+            return decodeRecoveryKey(recoveryKey);
+        }
+    };
+}
+
 async function getSecretStorageKey({ keys: keyInfos }, ssssItemName) {
```

Comment:
About your 4S prompt appearing when you don't want it to, which step is it that's accessing the 4S key precisely? This manager module only tries cached keys if we're inside an `accessSecretStorage` block, and there are many twisting code paths doing similar operations, so it may depend a lot on what operation is happening and when.